### PR TITLE
Emit warnings about deprecation as deprecation warnings

### PR DESCRIPTION
**Pull request by martijnhoekstra**
Move reporting to the extractor, and emit deprecation warnings that can be suppressed on supporting scala versions.

May also need including the Apache license, since the reporter was based on Apache licensed code, and a Notice file saying what and where, or just maybe hope for the best.

### CODE REVIEW

**Comment by martijnhoekstra** [comment]
is followup to #172

**Comment by SethTisue** [comment]
IANAL, but I don't think it's a large enough amount of code to trigger any licensing concerns. It's just glue.

**Comment by SethTisue** [comment]
@dwijnand 
1) mind reviewing this?
2) I guess this PR should be updated once 2.12.13 (any day now) ships, since it has the configurable warnings backport, and we shouldn't merge/publish this PR until then?

**Comment by martijnhoekstra** [comment]
O, right, I knew that was coming. If that comes with the overload to the reporter, I need to further segment the conditional scala versions.

**Comment by dwijnand** [comment]
Nothing concerning popped out the screen to me.

**Comment by martijnhoekstra** [comment]
updated for 2.12.13 to show the intended strategy. If it's ok, it can be quickly updated to 2.12.13 final.

**Comment by SethTisue** [comment]
2.12.13 is out

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,6 +30,7 @@ jobs:
           - 2.12.10
           - 2.12.11
           - 2.12.12
+          - 2.12.13-bin-5c9699d
```
Comment (martijnhoekstra) [inline .github/workflows/ci.yml:33]:
```suggestion
          - 2.12.13
```

File: build.sbt [hunk 2]
```diff
@@ -11,6 +12,7 @@ inThisBuild {
       "2.12.10",
       "2.12.11",
       "2.12.12",
+      "2.12.13-bin-5c9699d",
```
Comment (martijnhoekstra) [inline build.sbt:15]:
```suggestion
      "2.12.13",
```

**Review by martijnhoekstra — COMMENTED** [review]
update to 2.12.13 final

**Review by larsrh — APPROVED** [review]

**Comment by SethTisue** [comment]
I don't think the amount of code taken from Scala.js is enough to trigger any licensing concern.

**Review by SethTisue — APPROVED** [review]

**Comment by larsrh** [comment]
Thanks @martijnhoekstra for your contribution!

Do we need a new release for this?

**Comment by martijnhoekstra** [comment]
I think it needs a new release for 2.12.13 anyway, doesn't it?

**Comment by larsrh** [comment]
I already back-published that one with 0.11.2.

**Comment by martijnhoekstra** [comment]
I think there is enough reason to roll a release for this, but up to you, really

**Comment by larsrh** [comment]
Done.

**Pull request merged**